govc/cluster/rule/create.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rule
+
+import (
+	"context"
+	"flag"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type create struct {
+	*SpecFlag
+	*InfoFlag
+
+	vmhost       bool
+	affinity     bool
+	antiaffinity bool
+}
+
+func init() {
+	cli.Register("cluster.rule.create", &create{})
+}
+
+func (cmd *create) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.SpecFlag = new(SpecFlag)
+	cmd.SpecFlag.Register(ctx, f)
+
+	cmd.InfoFlag, ctx = NewInfoFlag(ctx)
+	cmd.InfoFlag.Register(ctx, f)
+
+	f.BoolVar(&cmd.vmhost, "vm-host", false, "Virtual Machines to Hosts")
+	f.BoolVar(&cmd.affinity, "affinity", false, "Keep Virtual Machines Together")
+	f.BoolVar(&cmd.antiaffinity, "anti-affinity", false, "Separate Virtual Machines")
+}
+
+func (cmd *create) Process(ctx context.Context) error {
+	if cmd.name == "" {
+		return flag.ErrHelp
+	}
+	return cmd.InfoFlag.Process(ctx)
+}
+
+func (cmd *create) Usage() string {
+	return "NAME..."
+}
+
+func (cmd *create) Description() string {
+	return `Create cluster rule.
+
+Rules are not enabled by default, use the 'enable' flag to enable upon creation or cluster.rule.change after creation.
+
+One of '-affinity', '-anti-affinity' or '-vm-host' must be provided to specify the rule type.
+
+With '-affinity' or '-anti-affinity', at least 2 vm NAME arguments must be specified.
+
+With '-vm-host', use the '-vm-group' flag combined with the '-host-affine-group' and/or '-host-anti-affine-group' flags.
```

Comment:
`affine` and `anti-affine`, are they typo?

Reply:
I just used the same names as the fields: http://pubs.vmware.com/vsphere-6-5/index.jsp#com.vmware.wssdk.apiref.doc/vim.cluster.VmHostRuleInfo.html